app/controllers/charges_controller.rb

```diff
@@ -137,15 +137,23 @@ def create_square_payment_request(
     payment = api_response.data.payment
     receipt_url = payment[:receipt_url]
 
+    # TODO (yong): Also query to find out the recipient
+    contact = Contact.find_or_create_by(email: email)
```

Comment:
What are your thoughts on putting this in a Service? maybe called ContactRegistration?

Reply:
I can do that another PR

Reply:
Sounds good should be something like ContactCreator to follow the or convention I wrongly started

BTW I saw in the other PR that people ask for email subscriptions so I’d imagine that we’d want the default to be false?

Reply:
hmmm thats a good question, I'm not sure whats the best way to model this. One idea is that if people don't subscribe, we just don't save any information for them at all in the DB, eliminating the need for a `is_subscribed` column.

Reply:
Good point but since we’d still be leveraging the tbl for donation information when we do create items, we’d still need it

I guess we can keep it to default false for now